Fix mixing of the @TestSecurity annotation with HTTP request credentials inside one test method (#41174)
quarkusio/quarkus: docs/src/main/asciidoc/security-testing.adoc

mechanism when none is defined), then Basic Auth needs to be enabled explicitly,
for example by setting `quarkus.http.auth.basic=true` or `%test.quarkus.http.auth.basic=true`.

Similarly, if it becomes necessary to test security features using both `@TestSecurity` and Bearer token authentication,
you can leverage both like in the example below:

[source, java]
----
@Test
@TestSecurity(user = "Bob")
public void testSecurityMetaAnnotation {
    RestAssured.given()
            .auth().oauth2(getTokenForUser("Alice")) <1>
            .get("hello")
            .then()
            .statusCode(200)
            .body(Matchers.is("Hello Alice"));
    RestAssured.given()
            .get("hello")
            .then()
            .statusCode(200)
            .body(Matchers.is("Hello Bob")); <2>
}

@Path("hello")
public static class HelloResource {

  @Inject
  SecurityIdentity identity;

  @Authenticated
  @GET
  public String sayHello() {
       return "Hello " + identity.getPrincipal().getName();
  }
}
----

> <1> Bearer token authentication mechanism is used, for we send a Bearer token with the HTTP request.

Reviewer: ```suggestion
<1> Bearer token authentication mechanism is used, because a Bearer access token is sent with the HTTP request.
```
